### PR TITLE
Remove UseExperimentalRetryJoin from reference templates

### LINKED ISSUE #2496: Remove UseExperimentalRetryJoin from reference templates
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

The deprecated `UseExperimentalRetryJoin` feature is now marked as deprecated by cluster-api:

- https://github.com/kubernetes-sigs/cluster-api/pull/6938

Let's remove this option from all capz reference and test templates, and documentation.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by sayantani11**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2496 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Remove UseExperimentalRetryJoin from reference templates
```

### CODE REVIEW

**Comment by sayantani11** [comment]
@jackfrancis Can you review the changes once. Cause I am still a bit unsure about it.

File: templates/flavors/aad/machine-deployment.yaml [hunk 1]
```diff
@@ -45,6 +45,7 @@ spec:
   template:
     spec:
       useExperimentalRetryJoin: true
```
Comment (jackfrancis) [inline templates/flavors/aad/machine-deployment.yaml:47]: Thanks for this submission!

What we actually need to do is to remove all instances of the line `useExperimentalRetryJoin: true` entirely. (And we'll need to remove the addition of the deprecation message.)

Once we've removed all of the `useExperimentalRetryJoin: true` lines from template reference files, run `make generate-flavors` to ensure that all of the rendered templates are updated.

Comment (sayantani11) [reply templates/flavors/aad/machine-deployment.yaml:47]: Got it! Will do

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/assign @jackfrancis

**Review by mboersma — APPROVED** [review]
/lgtm
/approve

**Pull request merged**